### PR TITLE
feat(checkbox): rename dark property to on-background for consistency

### LINKED ISSUE #38: Checkbox: Add `on-background` property and deprecate `dark` property
**The problem**
Many components can be asked to display inverse-style colors using an `on-background` property. The odd one out is Checkbox, which uses a `dark` property. The `on-background` property should be added and preferred instead, for both consistency and decoupling from equating the concept of "dark" to the concept of "on-background."

**The solution**
* Add an `on-background` property that mirrors the current behavior of `dark`
* Mark `dark` as deprecated

### LINKED ISSUE #39: Checkbox: Remove the `dark` property
Following #38, remove the deprecated `dark` property in a major release.

### PR DESCRIPTION
**Pull request by daneah**
**This change:** (check at least one)

- [ ] Adds a new feature
- [ ] Fixes a bug
- [x] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [x] Yes
- [ ] No

**Is the:** (complete all)

- [x] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [x] Test suite(s) passing?
- [x] Code coverage maximal?
- [x] Changeset added?

**What does this change address?**

Closes #38 — we'll only be including this in v10, so don't need to address.
Resolves #39 

**How does this change work?**
Rename and update usages of previous `dark` property with `on-background`.

**Additional context**
I added `.tool-versions`, a file used by [asdf](https://asdf-vm.com), to `.gitignore`.

### CODE REVIEW

File: packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx [hunk 1]
```diff
@@ -140,25 +140,25 @@ export const Template = (args) => (
...
 
 <Canvas withToolbar>
-  <Story name="Dark">
+  <Story name="On background">
     {() => {
       return (
         <div>
-          <div className="checkbox-example__container--dark">
-            <PharosCheckbox name="dark" dark>
+          <div className="checkbox-example__container--on-background">
```
Comment (Niznikr) [inline packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx:150]: Think you'll need to rename `checkbox-example__container--dark` defined in `pharos-site/static/styles/global.scss`

Comment (daneah) [reply packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx:150]: Hmm, must've been in the wrong directory when I searched for that.

**Review by Niznikr — APPROVED** [review]

**Review by SMQuazi — APPROVED** [review]

File: packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx [hunk 2]
```diff
@@ -140,25 +140,25 @@ export const Template = (args) => (
...
-            <PharosCheckbox name="dark" dark checked>
+          <div className="checkbox-example__container--on-background">
+            <PharosCheckbox name="on-background" onBackground checked>
               <span slot="label">Checked</span>
             </PharosCheckbox>
           </div>
-          <div className="checkbox-example__container--dark">
-            <PharosCheckbox name="indeterminate" dark indeterminate>
+          <div className="checkbox-example__container--on-background">
+            <PharosCheckbox name="indeterminate" on-background indeterminate>
```
Comment (SMQuazi) [inline packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx:161]:
```suggestion
            <PharosCheckbox name="indeterminate" onBackground indeterminate>
```

Comment (Niznikr) [reply packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx:161]: One on 151 too

File: packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx [hunk 3]
```diff
@@ -140,25 +140,25 @@ export const Template = (args) => (
...
 <Canvas withToolbar>
-  <Story name="Dark">
+  <Story name="On background">
     {() => {
       return (
         <div>
-          <div className="checkbox-example__container--dark">
-            <PharosCheckbox name="dark" dark>
+          <div className="checkbox-example__container--on-background">
+            <PharosCheckbox name="on-background" on-background>
```
Comment (Niznikr) [inline packages/pharos/src/components/checkbox/PharosCheckbox.react.stories.mdx:151]:
```suggestion
            <PharosCheckbox name="on-background" onBackground>
```

**Pull request merged**